src/main/java/org/mafagafogigante/dungeon/schema/rules/EnumJsonRule.java

```diff
@@ -0,0 +1,36 @@
+package org.mafagafogigante.dungeon.schema.rules;
+
+import com.eclipsesource.json.JsonValue;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+class EnumJsonRule<T extends Enum<T>> extends StringJsonRule {
+
+  private final List<Enum<T>> enumValues = new ArrayList<>();
```

Comment:
This is the only issue I see. We are performing linear searches on which we have to call toString() on the values. This approach can lead to quadratic behavior which can be a real issue when we start using the JSON validation code to validate mods as some of these enums can grow quite large with time.

Using a `HashSet<String>` should allow for linear lookups.